### PR TITLE
Gradient testing

### PR DESCRIPTION
**Pull request by VedPatwardhan**

### CODE REVIEW

File: ivy_tests/test_ivy/helpers.py [hunk 1]
```diff
@@ -519,6 +524,82 @@ def value_test(*, ret_np_flat, ret_from_np_flat, rtol=None, atol=1e-6):
...
+    container_flags,
+    rtol_: float = None,
+    atol_: float = 1e-06,
+    ground_truth_backend: str = "torch",
+):
+    def grad_fn(xs):
+        array_vals = [v for k, v in xs.to_iterator()]
+        args_writeable = ivy.copy_nest(args)
+        kwargs_writeable = ivy.copy_nest(kwargs)
+        ivy.set_nest_at_indices(args_writeable, args_idxs, array_vals)
```
Comment (djl11) [inline ivy_tests/test_ivy/helpers.py:550]: we are setting the exact same array values into both the `args` and the `kwargs`, this is not correct.

File: ivy_tests/test_ivy/helpers.py [hunk 2]
```diff
@@ -1114,6 +1195,28 @@ def test_function(
         ivy.unset_backend()
         raise e
     ivy.unset_backend()
+    # gradient test
+    if (
+        test_gradients
+        and not fw == "numpy"
+        and all(as_variable_flags)
+        and not any(container_flags)
+        and not any(native_array_flags)
```
Comment (djl11) [inline ivy_tests/test_ivy/helpers.py:1204]: native arrays should be fine actually I think, or am I missing something? Native arrays can also be variables.

**Review by djl11 — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers.py [hunk 3]
```diff
@@ -509,6 +509,11 @@ def get_ret_and_flattened_array(func, *args, **kwargs):
 
 
 def value_test(*, ret_np_flat, ret_from_np_flat, rtol=None, atol=1e-6):
+    if type(ret_np_flat) != np.ndarray:
+        ret_np_flat = np.asarray(ret_np_flat)
+    if type(ret_from_np_flat) != np.ndarray:
+        ret_from_np_flat = np.asarray(ret_from_np_flat)
+    assert ret_np_flat.shape[0] == ret_from_np_flat.shape[0]
```
[the remark this answers is not in the document]
Comment (VedPatwardhan) [reply ivy_tests/test_ivy/helpers.py]: Yeah, I made a mistake here, I've fixed it now.

File: ivy_tests/test_ivy/helpers.py [hunk 4]
```diff
@@ -509,6 +509,11 @@ def get_ret_and_flattened_array(func, *args, **kwargs):
 
 
 def value_test(*, ret_np_flat, ret_from_np_flat, rtol=None, atol=1e-6):
+    if type(ret_np_flat) != np.ndarray:
```
[the remark this answers is not in the document]
Comment (VedPatwardhan) [reply ivy_tests/test_ivy/helpers.py]: Yeah, I made a mistake here too, I've fixed it now.

File: ivy_tests/test_ivy/helpers.py (hunk 1 above)
Comment (VedPatwardhan) [reply ivy_tests/test_ivy/helpers.py]: Yeah, I made a mistake here too, I've fixed it now.

File: ivy_tests/test_ivy/helpers.py (hunk 2 above)
Comment (VedPatwardhan) [reply ivy_tests/test_ivy/helpers.py]: Actually, native arrays further exceed the deadline to reach nearly 1700ms, so now I've removed the condition and have further increased the deadline to 2000ms.

**Review by djl11 — APPROVED** [review]
lgtm!

**Pull request merged**